lib/spack/docs/packaging_guide.rst

```diff
@@ -1897,6 +1897,25 @@ command line to find installed packages or to install packages with
 particular constraints, and package authors can use specs to describe
 relationships between packages.
 
+.. warning::
+
+   Although ``depends_on`` can handle any aspect of Spack's spec syntax,
+   it currently cannot handle recursive dependencies. If the ``^`` sigil
+   appears in a ``depends_on`` statement, the concretizer will hang.
+   For example, something like:
+
+   .. code-block:: python
+
+      depends_on('mfem+cuda ^hypre+cuda', when='+cuda')
+
+   should be rewritten as:
+
+   .. code-block:: python
+
+      depends_on('mfem+cuda', when='+cuda')
+      depends_on('hypre+cuda', when='+cuda')
```

Comment:
I can also move this to Known Issues if we want. Not sure whether to consider this a "bug" or a "feature". Personally I would like to remove this constraint in @tgamblin's new concretizer.

Reply:
I think "Known issues" makes sense. Whatever is the decision (support or not `^` in `depends_on`) the concretizer shouldn't hang. It should either solve the constraints or report an error / warning.